qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -358,18 +387,115 @@ def test_dataset_with_no_experiment_raises(empty_temp_db):
                             ParamSpec("y", "numeric")])
 
 
-def test_database_upgrade(empty_temp_db):
-    connection = connect(qc.config["core"]["db_location"],
-                 qc.config["core"]["db_debug"])
-    userversion = get_user_version(connection)
-    if userversion != 0:
-        raise RuntimeError("trying to upgrade from version 0"
-                           " but your database is version"
-                           " {}".format(userversion))
-    sql = 'ALTER TABLE "runs" ADD COLUMN "quality"'
+def test_guid(dataset):
+    guid = dataset.guid
+    assert len(guid) == 36
+    parse_guid(guid)
```

Comment:
shall we test here that the `generate_guid` call that was used when dataset was created, used current values of location, sample name, workstation from the config, and the timestamp is "fairly recent/reasonable"?

Reply:
No, I think it is sufficient to have that tested in `test_guids.py`